internal/backend/local/test.go

```diff
@@ -1225,74 +1235,21 @@ func (runner *TestFileRunner) AddVariablesToConfig(config *configs.Config, varia
 	// First, take a backup of the existing configuration so we can easily
 	// restore it later.
 	currentVars := make(map[string]*configs.Variable)
-	for name, variable := range config.Module.Variables {
+	for name, variable := range run.ModuleConfig.Module.Variables {
 		currentVars[name] = variable
 	}
 
 	for name, value := range variables {
-		if _, exists := config.Module.Variables[name]; exists {
+		if _, exists := run.ModuleConfig.Module.Variables[name]; exists {
 			continue
 		}
 
-		config.Module.Variables[name] = &configs.Variable{
+		run.ModuleConfig.Module.Variables[name] = &configs.Variable{
 			Name:           name,
 			Type:           value.Value.Type(),
 			ConstraintType: value.Value.Type(),
 			DeclRange:      value.SourceRange.ToHCL(),
 		}
 	}
 
-	// We return a function that will reset the variables within the config so
-	// it can be used again.
-	return func() {
-		config.Module.Variables = currentVars
-	}
```

Comment:
The need for such reset has been eliminated by creating a copy of the variables and provider config for each run